Add a pinpointer clearing item for pirates

whatston3 opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->

Adds a bag of ionized bluespace dust, some magical consumable that, when used (sprinkled) on an item, will clear any pinpointer that is targeting it.

Adds this as a 2 DB item to the pirate uplink.  It has three uses, and provides _no direct feedback_ on whether the item was tracked by a pinpointer or not.  There's also a scanner that beeps if anything within range has a pinpointer.

Fixes cases where you'd have a rotated question mark on pinpointers.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->

Pinpointers have zero counterplay at the moment.  With this, we could re-enable pinpointers targeting people rather than on arbitrary items on their person.  I think it'd be fair to have a similar syndicate version for smooglers, but I don't know what a consumable pinpointer clearer would look like for the syndicate.

Rotated question marks: just looks _awful_.

## Technical details
<!-- Summary of code changes for easier review. -->

Pinpointer data isn't stored on the target, so we can't efficiently clear association - everything has to be done by querying every pinpointer that exists (awful).  I didn't want to duplicate the PinpointerSystem (though we _could_) - that would fix some of this, we could have a parent-child relationship between pointer and pointee.

## How to test
<!-- Describe a procedure to test this feature, along with expected output/behavior. -->

1. Spawn a box of raisins, a stack of steel sheets, three pinpointers, and a bag of ionized bluespace dust.
2. Point two of the pinpointers at the raisins, the third at the steel sheets.
3. Turn one of the raisin pinpointers off.
4. Use the dust on the raisins.  You should get a 5 second doAfter.
5. Once that completes, the two pinpointers that were pointing to the raisins should be clear, and show an unrotated question mark on their screen.
6. The pinpointer that was pointing to the steel should still be pointing at the steel.
7. Connect on another client, spawn another bag of bluespace dust.
8. Use the bluespace dust on your other character.  You should get a big popup telling you that they're sprinkling blue dust on you.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

The popup message when bluespace dust is being sprinkled on you.
![image](https://github.com/user-attachments/assets/30105446-48b9-40ad-9360-b1b2533bdc82)

A video (1:36) showing the first bits of the test procedure.  The section at the end where the raisin box is removed and the arrow rotates downward should be fixed (it should be a question mark) - there was no appearance update happening on the server's PinpointerSystem.
https://drive.google.com/file/d/1ib_ka8VM1u1W1wAbQmQ51tJY4l0H2CzD/view?usp=sharing

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
- [X] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [X] I confirm that AI tools were not used in generating the material in this PR.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl:
- add: Added a bag of bluespace dust to the pirate uplink, it costs 1 DB and will clear any pinpointer pointing at one item.
- fix: Pinpointers should no longer display rotated question marks.

Bonaout commented:
Hello, pirate player here! I like the idea of having a way to remove pinpointer tracking, but unfortunately this would be useful only in the very specific cases where you already know what item is being tracked. Making it pretty expensive for a single use item, and would rarely ever be used, not helping a lot in pinpointer counterplay!

Maybe having a few uses before the item is consumed could be nice?

Either way, I still think this is a nice addition, thank you for your work!

Masterscape commented:
I agree with Bonaout, I literally pinpointed a diagonal wall from a their ship to track the pirates about 2-3 shifts back. You would also need to add an item to detect pinpointed items. Something like a particle effect, geiger counter or just adding some flavor text to the entity's examination text.

whatston3 commented:
Did a bit of work on this.

I've added a component that tracks a pinpointer target (pinpointee?) with its pinpointer(s) tracking it, and added another purchaseable pinpointer scanner for the pirates.  Unlike the others (for now), this isn't immediately obvious.  It'll beep quicker up until you're 3 meters away from the object.  Ideally, you shouldn't be able to immediately pick out what has a pinpointer on it.  You can fuzz it, or make a quick guess, but you should be able to make mistakes if you're rushing.

I've made the bag of dust cost 2 DB for three charges.  The detector costs 2 DB and runs on batteries.

There's also a new visualizer for limited charges (why this wasn't a thing I do not know).  Video of the sand and the pinpointer in use (0:59): https://drive.google.com/file/d/1df9LZzb-a0zhfFQCpm1nMAmqfPn32yxb/view?usp=sharing
